app/models/digital_object/publish_target.rb

```diff
@@ -132,6 +132,9 @@ def unpublish_digital_object(digital_object, do_ezid_update)
     )
     if do_ezid_update && !digital_object.doi.blank?
       # We need to change the state of this ezid to :unavailable
+      # PSEUDO CODE
+      # begin - rescue around the following call, rescue Hyacinth::Exceptions::DataciteError
+      # Question: What to do if above exception is recued/caught?
       success = digital_object.change_doi_status_to_unavailable
       digital_object.errors.add(:ezid_response, "An error occurred while attempting to set this digital object's ezid status to 'unavailable'.") unless success
```

Comment:
This is pseudocode, but you probably want something like this:

```suggestion
      begin
        digital_object.change_doi_status_to_unavailable
      rescue YourExceptionType1, YourExceptionType2, etc. => e
        digital_object.errors.add(:datacite, e.message)
      end
      status to 'unavailable'.") unless success
```